src/ModelContextProtocol.AspNetCore/HttpServerTransportOptions.cs

```diff
@@ -26,9 +26,17 @@ public class HttpServerTransportOptions
     /// Represents the duration of time the server will wait between any active requests before timing out an
     /// MCP session. This is checked in background every 5 seconds. A client trying to resume a session will
     /// receive a 404 status code and should restart their session. A client can keep their session open by
-    /// keeping a GET request open. The default value is set to 2 minutes.
+    /// keeping a GET request open. The default value is set to 2 hours.
     /// </summary>
-    public TimeSpan IdleTimeout { get; set; } = TimeSpan.FromMinutes(2);
+    public TimeSpan IdleTimeout { get; set; } = TimeSpan.FromHours(2);
+
+    /// <summary>
+    /// The maximum number of idle sessions to track. This is used to limit the number of sessions that can be idle at once.
+    /// Past this limit, the server will log a critical error and terminate the oldest idle sessions even if they have not reached
+    /// their <see cref="IdleTimeout"/> until the idle session count is below this limit. Clients that keep their session open by
+    /// keeping a GET request open will not count towards this limit. The default value is set to 10,000 sessions.
+    /// </summary>
+    public int MaxIdleSessionCount { get; set; } = 10_000;
```

Comment:
Assuming that these are all being handled asynchronously, isn't the 10k default limit kind of conservative for a server?

Reply:
It's probably also worth calling out that I expect a lot of clients are going to keep a GET request open to handle unsolicited messages [1]. This active GET request will exclude the session from being classified as idle. So you could end up with far more than MaxIdleSessionCount concurrent sessions. They just can't all be idle.

[1]: I've already seen this with the Streamable HTTP version of MCP Inspector in PR at https://github.com/modelcontextprotocol/inspector/pull/339